app/controllers/spree/admin/payments_controller.rb

```diff
@@ -24,9 +24,12 @@ def new
       end
 
       def create
+        # Try to redeem VINE voucher first as we don't want to create a payment and complete
+        # the order if it fails
+        return redirect_to spree.admin_order_payments_path(@order) unless redeem_vine_voucher
```

Comment:
Is this relevant for local vouchers also? 
I guess we can assume if they are validated already, they're more likely to successfully redeem. But perhaps it's also possible that they could fail (maybe the voucher was added to cart, then redeemed on another device).

Reply:
Not an issue for local voucher, they are considered "redeemed" as soon as they are added to the order. 

> I guess we can assume if they are validated already, they're more likely to successfully redeem. But perhaps it's also possible that they could fail (maybe the voucher was added to cart, then redeemed on another device)

Indeed, the voucher remaining amount can be lower that what we want to redeem or even all spent even though it was still good when it was validated.